container-engine-lib/lib/backend_impls/kubernetes/kubernetes_kurtosis_backend/shared_helpers/shared_helpers.go

```diff
@@ -361,13 +366,52 @@ func GetUserServiceKubernetesResourcesMatchingGuids(
 				resultObj = &UserServiceKubernetesResources{
 					Service: nil,
 					Pod:     nil,
+					Ingress: nil,
 				}
 			}
 			resultObj.Pod = kubernetesPod
 			results[serviceUuid] = resultObj
 		}
 	}
 
+	// Get k8s ingresses
+	matchingKubernetesIngresses, err := kubernetes_resource_collectors.CollectMatchingIngresses(
+		ctx,
+		kubernetesManager,
+		namespaceName,
+		kubernetesResourceSearchLabels,
+		kubernetes_label_key.GUIDKubernetesLabelKey.GetString(),
+		postFilterLabelValues,
+	)
+	if err != nil {
+		return nil, stacktrace.Propagate(err, "An error occurred getting Kubernetes ingresses matching service UUIDs: %+v", serviceUuids)
+	}
+	for serviceGuidStr, kubernetesIngressForGuid := range matchingKubernetesIngresses {
+		logrus.Tracef("Found Kubernetes ingress for GUID '%v': %+v", serviceGuidStr, kubernetesIngressForGuid)
+		serviceUuid := service.ServiceUUID(serviceGuidStr)
+
+		numIngressesForGuid := len(kubernetesIngressForGuid)
+		if numIngressesForGuid == 0 {
+			// This would indicate a bug in our ingress retrieval logic because we shouldn't even have a map entry if there's nothing matching it
+			return nil, stacktrace.NewError("Got entry of result ingresses for service GUID '%v', but no Kubernetes ingress were returned; this is a bug in Kurtosis", serviceUuid)
+		}
+		if numIngressesForGuid > 1 {
+			return nil, stacktrace.NewError("Found %v Kubernetes ingresses associated with service GUID '%v'; this is a bug in Kurtosis", numIngressesForGuid, serviceUuid)
+		}
```

Comment:
nit: we could combined both check:
```suggestion
		if numIngressesForGuid != 1 {
			return nil, stacktrace.NewError("Found %v Kubernetes ingresses associated with service GUID '%v', but number of ingresses should be exactly 1", numIngressesForGuid, serviceUuid)
		}
```